### ISSUE TITLE
how does FConvLanguageModel mask future information?

### ISSUE THREAD

**Issue by scotthuang1989** [pytorch/fairseq#658]
**FConvLanguageModel**   is a language model base on CNN.  At training time, we will feed the whole sentence to the model, the target is the same sentence shift 1 position to right, it will cause model  use future information, so for language model, we need mask future information. But I can't find any code do that in this model.  Can someone help me out?

**Comment by huihuifan** [comment]
The convolutions are padded so they do not see future information. It’s an
argument you can pass to the convolutional layer.

**Comment by frankang** [comment]
here is the demo image for the language model part, from the paper _Convolutional Sequence to Sequence Learning_ 
https://arxiv.org/pdf/1705.03122.pdf
![image](https://user-images.githubusercontent.com/16100899/56725477-81efe180-677f-11e9-869d-6c7b49262894.png)

**Comment by scotthuang1989** [comment]
@huihuifan @frankang. thanks I am confused by the [paper](https://arxiv.org/pdf/1612.08083.pdf). in this paper it said the padding size is k/2 (k is the cnn kernel size), but the  padding size should be k -1.
